shell/platform/darwin/graphics/FlutterDarwinContextMetalSkia.mm

```diff
@@ -43,15 +46,19 @@ - (instancetype)initWithMTLDevice:(id<MTLDevice>)device
 
     [_commandQueue setLabel:@"Flutter Main Queue"];
 
-    CVReturn cvReturn = CVMetalTextureCacheCreate(kCFAllocatorDefault,  // allocator
-                                                  nil,      // cache attributes (nil default)
-                                                  _device,  // metal device
-                                                  nil,      // texture attributes (nil default)
-                                                  &_textureCache  // [out] cache
-    );
-    if (cvReturn != kCVReturnSuccess) {
-      FML_DLOG(ERROR) << "Could not create Metal texture cache.";
-      return nil;
+    {
+      CVMetalTextureCacheRef cache = nullptr;
+      CVReturn cvReturn = CVMetalTextureCacheCreate(kCFAllocatorDefault,  // allocator
+                                                    nil,      // cache attributes (nil default)
+                                                    _device,  // metal device
+                                                    nil,      // texture attributes (nil default)
+                                                    &cache    // [out] cache
+      );
+      if (cvReturn != kCVReturnSuccess) {
+        FML_DLOG(ERROR) << "Could not create Metal texture cache.";
+        return nil;
+      }
+      _textureCache.Reset(cache);
```

Comment:
Technically the docs say "When [the out param] value is NULL, an error occurred in texture creation."; on the offchance that there exists some edge case where Apple messed up and they populate the out param but *don't* return `kCVReturnSuccess`, we should probably put this line *before* the if block above so that we don't leak in the new version.

Reply:
Good call. Done.